src/test/java/seedu/address/logic/commands/LinkCommandTest.java

```diff
@@ -0,0 +1,171 @@
+package seedu.address.logic.commands;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandFailure;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.address.logic.commands.CommandTestUtil.showModuleClassAtIndex;
+import static seedu.address.logic.commands.CommandTestUtil.showStudentAtIndex;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_MODULE_CLASS;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_STUDENTS;
+import static seedu.address.testutil.Assert.assertThrows;
+import static seedu.address.testutil.TypicalIndexes.INDEX_FIRST_ITEM;
+import static seedu.address.testutil.TypicalIndexes.INDEX_SECOND_ITEM;
+import static seedu.address.testutil.TypicalTutorsPet.getNoLinkTutorsPet;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.UUID;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.moduleclass.ModuleClass;
+import seedu.address.model.student.Student;
+
+/**
+ * Contains integration tests (interaction with the Model) for {@code LinkCommand}.
+ */
+public class LinkCommandTest {
+
+    private Model model = new ModelManager(getNoLinkTutorsPet(), new UserPrefs());
+
+    @Test
+    public void constructor_nullStudentIndex_throwNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new LinkCommand(INDEX_FIRST_ITEM, null));
+    }
+
+    @Test
+    public void constructor_nullModuleClassIndex_throwNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new LinkCommand(null, INDEX_FIRST_ITEM));
+    }
+
+    @Test
+    public void execute_unfilteredList_success() {
+        // manually link first class to first student
+        ModuleClass moduleClass = model.getFilteredModuleClassList().get(0);
+        Student student = model.getFilteredStudentList().get(0);
+        Set<UUID> studentUuids = new HashSet<>(moduleClass.getStudentUuids());
+        studentUuids.add(student.getUuid());
+        ModuleClass modifiedModuleClass = new ModuleClass(moduleClass.getName(), studentUuids);
```

Comment:
would it be better to leave a line here?
```suggestion
       
         // manually link first class to first student
        ModuleClass moduleClass = model.getFilteredModuleClassList().get(0);
        Student student = model.getFilteredStudentList().get(0);
        Set<UUID> studentUuids = new HashSet<>(moduleClass.getStudentUuids());
        studentUuids.add(student.getUuid());
        ModuleClass modifiedModuleClass = new ModuleClass(moduleClass.getName(), studentUuids);
```

Reply:
I think for this case, it would be okay to not include an empty line above the comment since it is the first line in the method body. It will also be more in line with existing code.